src/main/java/org/jenkinsci/bytecode/Transformer.java

```diff
@@ -57,16 +58,28 @@ public synchronized void loadRules(Collection<? extends ClassLoader> loaders) th
             return image;
 
         final ClassReader cr = new ClassReader(image);
-        final ClassWriter cw = new ClassWriter(/*ClassWriter.COMPUTE_FRAMES|*/ClassWriter.COMPUTE_MAXS);
+        final ClassWriter cw = new ClassWriter(ClassWriter.COMPUTE_FRAMES|ClassWriter.COMPUTE_MAXS);
 
         final boolean[] modified = new boolean[1];
 
-        cr.accept(new ClassVisitor(ASM5,cw) {
+        // If code contains JSR/RET instructions then ASM fails to transform it with
+        // java.lang.RuntimeException: JSR/RET are not supported with computeFrames option
+        // so inline any JSR subroutines
```

Comment:
@jglick yes - so this will now apply a larger transformation than previously if some is deemed required.

I can go with @kohsuke's suggestion which is not too much extra work - but I'm not sure I see the benefit - especially as we are already potentially up-versioning some byte-code.
>  We could have chosen to set the COMPUTE_FRAMES flag based on class file version, or we can just inline JSR/RET automatically, and I guess we opted for the latter.